### PR TITLE
feat: use order watcher service

### PR DESCRIPTION
**Pull request by z2trillion**
# Description

# Testing Instructions

# Checklist

-   [x] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [x] Prefix PR title with `[WIP]` if necessary.
-   [x] Add tests to cover changes as needed.
-   [ ] Test changes on the staging environment with [Matcha API Staging](https://api-staging.matcha.xyz).

    -   [ ] SRA/Limit orders
    -   [ ] Swap endpoints
    -   [ ] Meta transaction endpoints
    -   [ ] Depth charts

    For more information see `0x API Matcha smoke test runbook` in Quip.

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -187,31 +187,20 @@ export async function getDefaultAppDependenciesAsync(
...
-    // Register a shutdown event listener.
-    // TODO: More teardown logic should be added here. For example, the mesh rpc
-    // client should be destroyed and services should be torn down.
+    const { server } = await runHttpServiceAsync(dependencies, config, app);
+
     server.on('close', async () => {
-        await wsService.destroyAsync();
+        // Register a shutdown event listener.
+        // TODO: More teardown logic should be added here. For example, the mesh rpc
+        // client should be destroyed and services should be torn down.
```
Comment (opaolini) [inline src/app.ts:203]:
```suggestion
        // TODO: More teardown logic should be added here. For example individual services should be torn down.
```

Comment (opaolini) [reply src/app.ts:203]: Just part of removing Mesh from the codebase.

File: src/config.ts [hunk 2]
```diff
@@ -173,6 +173,22 @@ export const DB_ORDERS_UPDATE_CHUNK_SIZE = 300;
...
+    ? undefined
+    : assertEnvVarType('KAFKA_BROKERS', process.env.KAFKA_BROKERS, EnvVarType.StringList);
+
+export const KAFKA_CONSUMER_GROUP_ID = _.isEmpty(process.env.KAFKA_CONSUMER_GROUP_ID)
+    ? undefined
+    : assertEnvVarType('KAFKA_CONSUMER_GROUP_ID', process.env.KAFKA_CONSUMER_GROUP_ID, EnvVarType.NonEmptyString);
+
 // Mesh Endpoint
 export const MESH_WEBSOCKET_URI = _.isEmpty(process.env.MESH_WEBSOCKET_URI)
     ? undefined
```
Comment (opaolini) [inline src/config.ts:194]: Maybe we should delete all of this?

File: src/utils/order_watcher_utils.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import { OrderEventEndState } from '@0x/mesh-graphql-client';
```
Comment (opaolini) [inline src/utils/order_watcher_utils.ts:1]: I think we should remove it, explicitly import it and remove mesh dependencies altogether.

File: test/orderbook_service_test.ts [hunk 4]
```diff
@@ -3,24 +3,25 @@ import { BlockchainLifecycle, Web3ProviderEngine, Web3Wrapper } from '@0x/dev-ut
 import { OrderEventEndState } from '@0x/mesh-graphql-client';
 import { LimitOrderFields } from '@0x/protocol-utils';
 import { BigNumber } from '@0x/utils';
+// import axios from 'axios';
+// import AxiosMockAdapter from 'axios-mock-adapter';
```
Comment (opaolini) [inline test/orderbook_service_test.ts:7]:
```suggestion
```

File: src/services/websocket_service.ts [hunk 5]
```diff
@@ -19,11 +20,17 @@ import {
...
+import { OrderWatcherEvent, orderWatcherEventToSRAOrder } from '../utils/order_watcher_utils';
 import { schemaUtils } from '../utils/schema_utils';
 
+const ID_CEILING = 10000;
+
+const getRandomKafkaConsumerGroupId = (): string => {
+    const randomInt = Math.floor(Math.random() * ID_CEILING);
+    return `sra_0x_api_service_${randomInt}`;
+};
+
```
Comment (opaolini) [inline src/services/websocket_service.ts:33]: I'll change this to random hexbytes.

**Review by opaolini — COMMENTED** [review]
Outside the few comments, there's still a few things that we should probably include in this PR:
* Remove mesh variables from `config.ts`
* Remove `mesh_utils` and / or move them
* Remove mesh configs 
* Remove `/test/meta_transaction_test` ? 
* Remove mesh from `/test/docker-compose.yml`
* Remove Mesh imports from `/test/orderbook_service_test.ts`
* Remove Mesh mentionds in the main README.md

**Comment by z2trillion** [comment]
I verified that the order gen script for posting persistent orders still works.

**Pull request merged**